client/src/components/Home/Media/SpotifyPlayer.jsx

```diff
@@ -91,27 +143,175 @@ const SpotifyPlayer = ({ token, trackUri }) => {
     });
   };
 
-  // Handle play/pause button click
   const handlePlayPause = () => {
+    if (!deviceId) {
+      console.error("Device ID is not set");
+      return;
+    }
     if (isPaused) {
-      play({
-        playerInstance: player,
-        spotify_uri: trackUri,
-      });
+      play({ playerInstance: player, spotify_uri: trackUri });
     } else {
       player.pause();
     }
   };
 
+  const handleNextTrack = () => {
+    if (player) {
+      player.nextTrack();
+    }
+  };
+
+  const handlePreviousTrack = () => {
+    if (player) {
+      player.previousTrack();
+    }
+  };
+
+  const handleVolumeChange = (e) => {
+    const newVolume = e.target.value;
+    setVolume(newVolume);
+    if (player) {
+      player.setVolume(newVolume / 100);
+    }
+  };
+
+  const handlePlaylistSelection = (playlistId) => {
+    setSelectedPlaylists((prevSelected) =>
+      prevSelected.includes(playlistId)
+        ? prevSelected.filter((id) => id !== playlistId)
+        : [...prevSelected, playlistId]
+    );
+  };
+
+  const handleAddToPlaylists = () => {
+    if (!trackUri || selectedPlaylists.length === 0) return;
+
+    selectedPlaylists.forEach((playlistId) => {
+      fetch(`https://api.spotify.com/v1/playlists/${playlistId}/tracks`, {
+        method: "POST",
+        headers: {
+          Authorization: `Bearer ${token}`,
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ uris: [trackUri] }),
+      })
+        .then((response) => {
+          if (response.ok) {
+            console.log(`Track added to playlist ${playlistId}`);
+          } else {
+            console.error(`Error adding track to playlist ${playlistId}`);
+          }
+        })
+        .catch((error) => {
+          console.error(`Error adding track to playlist ${playlistId}`, error);
+        });
+    });
+    setShowPlaylists(false);
+  };
+
+  const formatTime = (milliseconds) => {
+    const minutes = Math.floor(milliseconds / 60000);
+    const seconds = ((milliseconds % 60000) / 1000).toFixed(0);
+    return seconds == 60
+      ? minutes + 1 + ":00"
+      : minutes + ":" + (seconds < 10 ? "0" : "") + seconds;
+  };
+
   return (
```

Comment:
It is very impressive you build the spotify player by yourself from scratch! Good work Frank